app/services/activitypub/process_status_update_service.rb

```diff
@@ -266,11 +264,12 @@ def broadcast_updates!
   def queue_poll_notifications!
     poll = @status.preloadable_poll
 
-    # If the poll had no expiration date set but now has, and people have
-    # voted, schedule a notification
+    # If the poll had no expiration date set but now has, or now has a sooner
+    # expiration date, and people have voted, schedule a notification
 
-    if @previous_expires_at.nil? && poll.present? && poll.expires_at.present? && poll.votes.exists?
-      PollExpirationNotifyWorker.perform_at(poll.expires_at + 5.minutes, poll.id)
-    end
+    return unless poll.present? && poll.expires_at.present? && poll.votes.exists?
+
+    PollExpirationNotifyWorker.remove_from_scheduled(poll.id) if @previous_expires_at.present?
```

Comment:
I think I would do that only if the `@previous_expires_at.present? && @previous_expires_at > poll.expires_at`, to avoid going through the sidekiq scheduled set after each update (since updates are a normal and possibly frequent mechanism to update votes count).